File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPAbsorptionAmount.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPAbsorptionAmount implements PlayerStateProperty<Double> {
    @Override
    public String getConfigName() {
        return "absorption_amount";
    }

    @Override
    public Double getDefaultSerializedProperty() {
        return 0D;
    }

    @Override
    public Double getSerializedPropertyFromPlayer(Player player) {
        return player.getAbsorptionAmount();
    }

    @Override
    public void restoreProperty(Player player, Double absorption) {
        player.setAbsorptionAmount(absorption);
    }

    @Override
    public Class<Double> getTypeClass() {
        return Double.class;
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPAbsorptionAmount.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.configuration.ConfigurationSection
import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPAbsorptionAmount extends PlayerStateProperty[Double]:
  override def getConfigName: String = "absorption_amount"

  override def getDefaultSerializedProperty: Double = 0D

  override def getSerializedPropertyFromPlayer(player: Player): Double =
    player.getAbsorptionAmount

  override def restoreProperty(player: Player, absorption: Double): Unit =
    player.setAbsorptionAmount(absorption)

  override def getValueFromConfig(config: ConfigurationSection): Double =
    config.getDouble(getConfigName)